### ISSUE TITLE
Notes on minor bugs, fixes, and training results

### ISSUE THREAD

**Issue by RLingHan** [1KK077/IDKL#14]
I tried running the code myself but encountered quite a few bugs. I eventually managed to fix them, so before running the code, I recommend making the following changes:
1. In baseline.py, when you try to import losses such as `TripletLoss` or `RerankLoss`, make sure to modify the import statements to something like:
   "from layers.loss.triplet_loss import TripletLoss"
   (and similarly for the other)
2. In baseline.py, starting from around line 349, the definitions like
  "self.shared_module_fr = Shared_module_fr"
   and the following lines are missing the "modality_attention` parameter. You’ll need to locate their corresponding constructors and set a default value for modality_attention, =0.
3. In the __init__() method of baseline.py, add the following line:
   self.mutual_learning = kwargs.get('mutual_learning', False)
4. For some reasons, I modified the engine code to avoid using Ignite and Apex — instead, I used PyTorch’s built-in AMP for mixed precision training.
5. If you want to view evaluation results during training, you’ll need to adjust the start_eval and eval_interval in the .yml configuration file. I set them both to 5.
I hope these notes are helpful. I’m just a beginner who has only been learning for about a month, so please feel free to correct me if anything above is inaccurate. Thank you!

As for the SYSU-MM01 dataset, I only trained for 100 epochs with rerank = False.
Here are the results for the last 10 epochs:

Epoch[91] Batch [150/195]	Speed: 75.86 samples/sec	F_p: 0.0000	IN_p: 0.0000	acc: 0.9968	bg_kl: 0.3500	ce: 0.3609	id_pl: 0.3818	pse: 0.6931	sm_kl: 0.4422	soft_dt: 0.1010	tri: 2.1906	unad: 0.0065
Epoch 91 finished in 307.23s
Epoch[92] Batch [150/195]	Speed: 75.83 samples/sec	F_p: 0.0000	IN_p: 0.0000	acc: 0.9965	bg_kl: 0.3501	ce: 0.3596	id_pl: 0.3800	pse: 0.6931	sm_kl: 0.4395	soft_dt: 0.1022	tri: 2.1816	unad: 0.0065
Epoch 92 finished in 307.34s
Epoch[93] Batch [150/195]	Speed: 75.85 samples/sec	F_p: 0.0000	IN_p: 0.0000	acc: 0.9964	bg_kl: 0.3672	ce: 0.3701	id_pl: 0.3923	pse: 0.6931	sm_kl: 0.4570	soft_dt: 0.1026	tri: 2.1972	unad: 0.0068
Epoch 93 finished in 307.10s
Epoch[94] Batch [150/195]	Speed: 75.94 samples/sec	F_p: 0.0000	IN_p: 0.0000	acc: 0.9966	bg_kl: 0.3543	ce: 0.3596	id_pl: 0.3809	pse: 0.6931	sm_kl: 0.4455	soft_dt: 0.1044	tri: 2.1812	unad: 0.0063
Epoch 94 finished in 307.02s
Epoch[95] Batch [150/194]	Speed: 76.03 samples/sec	F_p: 0.0000	IN_p: 0.0000	acc: 0.9972	bg_kl: 0.3425	ce: 0.3556	id_pl: 0.3765	pse: 0.6931	sm_kl: 0.4331	soft_dt: 0.1063	tri: 2.1706	unad: 0.0064
Epoch 95 finished in 305.12s
Model saved to checkpoints/sysu/SYSU/SYSU_model_95.pth

all num-shot:1 r1 precision = 69.65 , r10 precision = 95.73 , r20 precision = 98.67, mAP = 68.32
Epoch[96] Batch [150/195]	Speed: 76.05 samples/sec	F_p: 0.0000	IN_p: 0.0000	acc: 0.9968	bg_kl: 0.3458	ce: 0.3513	id_pl: 0.3721	pse: 0.6931	sm_kl: 0.4397	soft_dt: 0.1091	tri: 2.1639	unad: 0.0063
Epoch 96 finished in 307.12s
Epoch[97] Batch [150/195]	Speed: 76.12 samples/sec	F_p: 0.0000	IN_p: 0.0000	acc: 0.9971	bg_kl: 0.3526	ce: 0.3521	id_pl: 0.3728	pse: 0.6931	sm_kl: 0.4408	soft_dt: 0.1099	tri: 2.1677	unad: 0.0065
Epoch 97 finished in 306.37s
Epoch[98] Batch [150/195]	Speed: 76.04 samples/sec	F_p: 0.0000	IN_p: 0.0000	acc: 0.9971	bg_kl: 0.3461	ce: 0.3511	id_pl: 0.3723	pse: 0.6931	sm_kl: 0.4315	soft_dt: 0.1094	tri: 2.1651	unad: 0.0064
Epoch 98 finished in 306.46s
Epoch[99] Batch [150/195]	Speed: 76.08 samples/sec	F_p: 0.0000	IN_p: 0.0000	acc: 0.9969	bg_kl: 0.3574	ce: 0.3550	id_pl: 0.3772	pse: 0.6931	sm_kl: 0.4484	soft_dt: 0.1090	tri: 2.1720	unad: 0.0064
Epoch 99 finished in 306.44s
Epoch[100] Batch [150/195]	Speed: 76.31 samples/sec	F_p: 0.0000	IN_p: 0.0000	acc: 0.9969	bg_kl: 0.3461	ce: 0.3483	id_pl: 0.3691	pse: 0.6931	sm_kl: 0.4388	soft_dt: 0.1098	tri: 2.1558	unad: 0.0065
Epoch 100 finished in 305.78s
Model saved to checkpoints/sysu/SYSU/SYSU_model_100.pth

all num-shot:1 r1 precision = 69.70 , r10 precision = 95.64 , r20 precision = 98.54, mAP = 68.35
Training completed. Best Rank1: 70.3576

**Comment by lqf-wq** [comment]
你好，貌似在这份代码里面我没有看到测试代码，还有就是训练这份代码大概需要多少时间呢

**Comment by RLingHan** [comment]
> 你好，貌似在这份代码里面我没有看到测试代码，还有就是训练这份代码大概需要多少时间呢

在utils下，大概8到12个小时，因机器而异

**Comment by GuHY777** [comment]
最终大家有没有成果跑出论文的结果呢？
